### PR TITLE
gazelle: Add target for updating from go.mod

### PR DESCRIPTION
**Pull request by minor-fixes**
This change replaces the mystery command that I keep forgetting with a
bazel run target, making it easier than ever to generate
go_repositories.bzl from go.mod.

Tested: `bazel run //:gazelle_update_repos`

### CODE REVIEW

File: bazel/go_repositories.bzl [hunk 1]
```diff
@@ -2560,13 +2560,6 @@ def go_repositories():
         version = "v0.2.1",
     )
 
-    go_repository(
-        name = "com_github_xenking_zipstream",
-        importpath = "github.com/xenking/zipstream",
-        sum = "h1:6LfcpXfxO9kAGi0a+2N5C0ZZ6jyG4XULPiogOM7gJBU=",
-        version = "v1.0.1",
-    )
```
Comment (gpaussaenf) [inline bazel/go_repositories.bzl:2568]: Why was this removed? It's currently being referenced by enkit/bestie/server/test_result.go.

**Review by gpaussaenf — COMMENTED** [review]
Need to retain the xenking/zipstream package for use by bestie.

**Review by ccontavalli — APPROVED** [review]
Please address Greg's comment, otherwise looks fine.

File: bazel/go_repositories.bzl (hunk 1 above)
Comment (minor-fixes) [reply bazel/go_repositories.bzl]: It wasn't added to go.mod. The steps for adding a dep are:

* Add it to go.mod with: `go get -u $DEP_URL`
* Update `go_repositories.bzl` with: `bazel run //:gazelle_update_repos`

so go.mod is the source-of-truth, and manual edits to go_repositories.bzl will be discarded on the next run.

Added zipstream to go.mod to fix this issue
